### ISSUE TITLE
Private email relays

### ISSUE THREAD

**Issue by phuze** [nicokaiser/passport-apple#79]
When a standard user signs-in using Apple Login, they are given the option to either share or hide their email.

![apple-share](https://github.com/nicokaiser/passport-apple/assets/9218035/196b7d07-a76e-4905-a9bc-54b15eaa8275)

If they choose to hide their email, Apple will instead return a private relay email. That email address can be used to email the user, without exposing their email address.

In my application scenario, we require the users actual email address. Its perfectly fine if the users chooses to hide their email -- we just notify them that they cannot use Apple Login under those conditions.

When a user chooses to hide their email, Apple returns a `is_private_email` property, as part of the token payload. Examples:

User chose to **hide** their email:
```json
{
    "iss": "https://appleid.apple.com",
    "aud": "com.batman.auth",
    "exp": 1684529111,
    "iat": 1684442711,
    "sub": "111111.x111x1x11x111xx1x11111x1x11xx11x.1111",
    "at_hash": "XXXxXxx11xxxXXXxxX-XXX",
    "email": "x1x1xxx11x@privaterelay.appleid.com",
    "email_verified": "true",
    "is_private_email": "true",
    "auth_time": 1684442710,
    "nonce_supported": true
}
```

User chose to **share** their email:
```json
{
    "iss": "https://appleid.apple.com",
    "aud": "com.batman.auth",
    "exp": 1684525825,
    "iat": 1684439425,
    "sub": "111111.x111x1x11x111xx1x11111x1x11xx11x.1111",
    "at_hash": "XXxxXx1XxxXxx1xxxXxx_X",
    "email": "my.email@domain.com",
    "email_verified": "true",
    "auth_time": 1684439424,
    "nonce_supported": true
}
```

While I suppose we could just parse the returned email, looking for a substring `privaterelay` -- but this isn't ideal. We cannot guarantee that Apple will always maintain this format, and for all users.

Instead, the returned user profile should be updated to include `is_private_email`, should it exist. This provides an assured way of knowing whether the user chose to share, or hide, their email address.

**Comment by nicokaiser** [comment]
Nice catch, thanks! Maybe the property should be named `isPrivateEmail` instead of `privateEmail` to be consistent with the data received in the JWT, what do you think?

**Comment by phuze** [comment]
Make's sense. I've updated my PR with that [change](https://github.com/nicokaiser/passport-apple/pull/80/commits/11658c0ce5430cc482d7ef623e63b3a5635c416c). Cheers.

**Comment by nicokaiser** [comment]
Merged the PR and released 3.1.0. Thanks for your contribution!
